homeassistant/config_entries.py

```diff
@@ -101,8 +101,8 @@ class ConfigEntryState(Enum):
     def __new__(cls: type[object], value: str, recoverable: bool) -> ConfigEntryState:
         """Create new ConfigEntryState."""
         obj = object.__new__(cls)
-        obj._value_ = value
-        obj._recoverable = recoverable
+        obj._value_ = value  # type: ignore[attr-defined]
+        obj._recoverable = recoverable  # type: ignore[attr-defined]
```

Comment:
Those ignores are not necessary if we improve the method signature. Instead of `type[object]` for `cls`, `__new__` should accept a type of `TypeVar` bound to `ConfigEntryState`. That will make the cast unnecessary, too.

```py
T = TypeVar("T", bound="ConfigEntryState")

class ConfigEntryState(Enum):
    ...
    def __new__(cls: type[T], value: str, recoverable: bool) -> T:
        ...
        return obj
```

Once PEP 673 gets accepted, this can be simplified further: https://www.python.org/dev/peps/pep-0673/
For now, this only works in Pylance.
```py
from typing_extensions import Self

class ConfigEntryState(Enum):
    ...
    def __new__(cls, value: str, recoverable: bool) -> Self:
        ...
        return obj
```

Reply:
It would be great if this could be down in a separate PR. Will make it easier for other contributors to review core code. The change should already work with the current 0.910.